File: freecodecamp/daily-code-challenge/2026-03-14.js
function getPiDecimal(n) {
  let target = n + 1;

  let q = 1n, r = 0n, t = 1n, k = 1n, n1 = 3n, l = 3n;
  let digits = [];

  while (digits.length < target) {
    if (4n*q + r - t < n1*t) {
      digits.push(Number(n1));

      let newQ = 10n*q;
      let newR = 10n*(r - n1*t);
      let newN = ((10n*(3n*q + r)) / t) - 10n*n1;

      q = newQ;
      r = newR;
      n1 = newN;
    } else {
      let newQ = q*k;
      let newR = (2n*q + r)*l;
      let newT = t*l;
      let newK = k + 1n;
      let newN = (q*(7n*k) + 2n + r*l) / (t*l);
      let newL = l + 2n;

      q = newQ;
      r = newR;
      t = newT;
      k = newK;
      n1 = newN;
      l = newL;
    }
  }

  return digits[n];
}

File: freecodecamp/daily-code-challenge/2026-03-14.py
def get_pi_decimal(n):
    target = n + 1
    
    q, r, t, k, n1, l = 1, 0, 1, 1, 3, 3
    digits = []
    
    while len(digits) < target:
        if 4*q + r - t < n1*t:
            digits.append(n1)
            q, r, t, k, n1, l = (
                10*q,
                10*(r - n1*t),
                t,
                k,
                ((10*(3*q + r)) // t) - 10*n1,
                l
            )
        else:
            q, r, t, k, n1, l = (
                q*k,
                (2*q + r)*l,
                t*l,
                k + 1,
                (q*(7*k) + 2 + r*l) // (t*l),
                l + 2
            )
    
    return digits[n]
